src/test/scala/firrtl/backends/experimental/smt/end2end/EndToEndSMTSpec.scala

```diff
@@ -167,42 +169,32 @@ private object Z3ModelChecker extends LazyLogging {
     version
   }
 
-  def bmc(testDir: File, main: String, kmax: Int): MCResult = {
+  def bmc(testDir: Path, main: String, kmax: Int): MCResult = {
     assert(kmax >= 0 && kmax < 50, "Trying to keep kmax in a reasonable range.")
-    val smtFile = new File(testDir, main + ".smt2")
-    val header = read(smtFile)
-    val steps = (0 to kmax).map(k => new File(testDir, main + s"_step$k.smt2")).zipWithIndex
-    steps.foreach {
-      case (f, k) =>
-        writeStep(f, main, header, k)
-        val success = executeStep(f.getAbsolutePath)
-        if (!success) return MCFail(k)
+    Seq.tabulate(kmax + 1) { k =>
```

Comment:
`tabulate` returns a sequence, the `foreach` made more sense, since this execution is all about its side effects

Reply:
done.